tequilapi/endpoints/transactor.go

```diff
@@ -427,6 +427,15 @@ func (te *transactorEndpoint) Withdraw(c *gin.Context) {
 	}
 
 	chainID := config.GetInt64(config.FlagChainID)
+	if req.ChainID != 0 {
+		if _, ok := registry.Chains()[req.ChainID]; !ok {
+			utils.SendError(resp, errors.New("Unsupported chain"), http.StatusBadRequest)
+			return
+		}
+
+		chainID = req.ChainID
+	}
+
 	err = te.promiseSettler.Withdraw(chainID, identity.FromAddress(req.ProviderID), common.HexToAddress(req.HermesID), common.HexToAddress(req.Beneficiary))
```

Comment:
this chain ID determines only from which chain to withdraw, therefore your changes do not accomplish what you want them to accomplish.

You'll need changes to `func (aps *hermesPromiseSettler) Withdraw(chainID int64, providerID identity.Identity, hermesID, beneficiary common.Address) error`. 

The method probably has to include two chain ids: to and from. Currently, the method internally uses:

`aps.config.L1ChainID` -> the chain to withdraw to
`chainID` -> the chainID that was passed as the chain that the withdrawal is originating from.